### PR TITLE
ENH run benchmarks from any folders

### PR DESCRIPTION
**Pull request by tomMoral**
With this PR, the `benchopt` client should be provided with a folder in which the benchmark is defined:

```bash
benchopt run benchmarks/lasso
```

### CODE REVIEW

File: benchmarks/lasso/solvers/cvxpy.py [hunk 1]
```diff
@@ -26,13 +22,12 @@ def set_objective(self, X, y, lmbd):
         self.problem = cp.Problem(cp.Minimize(
             loss + self.lmbd * cp.norm(self.beta, 1)))
 
+        # Hack cvxpy to be able to retrieve a sub optimal solution when
```
Comment (agramfort) [inline benchmarks/lasso/solvers/cvxpy.py:25]:
```suggestion
        # Hack cvxpy to be able to retrieve a suboptimal solution when
```

File: benchopt/base.py [hunk 2]
```diff
@@ -88,19 +95,18 @@ def is_installed(cls, env_name=None, raise_on_not_installed=None):
...
-                    exc_type, value, tb = module.import_ctx.import_error
+                    exc_type, value, tb = cls._import_ctx.import_error
                     raise exc_type(value).with_traceback(tb)
                 return False
             else:
                 return True
         else:
             return _run_shell_in_conda_env(
-                f"benchopt check-install {cls.benchmark} {cls.name}",
+                f"benchopt check-install {cls._module_filename} "
```
Comment (agramfort) [inline benchopt/base.py:108]: can you document somewhere how to use this? I don't manage to guess with check-install -h

File: benchopt/runner.py [hunk 3]
```diff
@@ -28,35 +28,59 @@
...
+
+    Parameters
+    ----------
+    objective : instance of BaseObjective
+        The objective to minimize.
+    solver : instance of BaseSolver
+        The solver to use.
+    meta : dict
+        Metadata passed to instantiate a Cost.
+        Contains objective, data, scale, XXX
```
Comment (agramfort) [inline benchopt/runner.py:67]: fix doc.

File: benchopt/runner.py [hunk 4]
```diff
@@ -28,35 +28,59 @@
...
+    sample : int | float
```
Comment (agramfort) [inline benchopt/runner.py:68]: we rename this to stopping ?

File: benchopt/runner.py [hunk 5]
```diff
@@ -96,34 +160,66 @@ def run_one_sample(objective, solver_class, solver_parameters, meta, sample,
...
+    benchmark : str
+        The path to the benchmark files.
+    objective : instance of BaseObjective
+        The objective to minimize.
+    solver : instance of BaseSolver
+        The solver to use.
+    meta : dict
+        Metadata passed to instanciate a Cost.
+        Contains objective, data, scale, XXX
+    max_samples : int
```
Comment (agramfort) [inline benchopt/runner.py:178]: max_runs ?

File: continuous_integration/install.sh [hunk 6]
```diff
@@ -7,6 +7,8 @@ UNAMESTR=`uname`
 TO_INSTALL=$(cat dev-requirements.txt)
 TO_INSTALL="python=$VERSION_PYTHON pip $TO_INSTALL"
 
+# conda update -n base -c defaults conda -y
```
Comment (agramfort) [inline continuous_integration/install.sh:10]: clean up

**Review by tomMoral — COMMENTED** [review]
Ready for merge I think.

File: benchopt/base.py [hunk 7]
```diff
@@ -193,29 +206,29 @@ def __init__(self, **parameters):
...
 
         This function should not return the parameters which will be
         retrieved by a subsequent call to get_result.
 
         Parameters
         ----------
-        n_iter : int
-            Number of iteration to run the solver for. It allows to sample the
-            time/accuracy curve in the benchmark.
+        stop_val : int
```
Comment (agramfort) [inline benchopt/base.py:229]:
```suggestion
        stop_val : int | float
```

**Review by agramfort — APPROVED** [review]
just a nitpick.

let's merge this beast !

**Comment by agramfort** [comment]
🍻

**Pull request merged**